scijava/scijava-ops-engine/src/main/java/org/scijava/ops/engine/matcher/simplify/PrimitiveArraySimplifiers.java

```diff
@@ -44,63 +45,106 @@
 		.stream().map(i -> i == null ? null : i.intValue()).toArray(Integer[]::new);
 
-	@OpHints(hints = {Simplification.FORBIDDEN})
+	@OpHints(hints = { Simplification.FORBIDDEN })
 	@OpField(names = "simplify")
 	public final Function<Short[], ObjectArray<Number>> shortArrSimplifier =
-		s -> new ObjectArray<>(s);
-
-	@OpHints(hints = {Simplification.FORBIDDEN})
+		arr -> {
+			var oa = new ObjectArray<>(Number.class, arr.length);
+			// TODO: Why doesn't System.arraycopy work?
+			for(int i = 0; i < arr.length; i++) {
+				oa.set(i, arr[i]);
+			}
+			return oa;
+		};
+
+	@OpHints(hints = { Simplification.FORBIDDEN })
 	@OpField(names = "focus")
 	public final Function<ObjectArray<Number>, Short[]> shortArrFocuser = o -> o
 		.stream().map(s -> s == null ? null : s.shortValue()).toArray(Short[]::new);
 
-	@OpHints(hints = {Simplification.FORBIDDEN})
+	@OpHints(hints = { Simplification.FORBIDDEN })
 	@OpField(names = "simplify")
 	public final Function<Long[], ObjectArray<Number>> longArrSimplifier =
-		l -> new ObjectArray<>(l);
-
-	@OpHints(hints = {Simplification.FORBIDDEN})
+		arr -> {
+			var oa = new ObjectArray<>(Number.class, arr.length);
+			// TODO: Why doesn't System.arraycopy work?
+			for(int i = 0; i < arr.length; i++) {
+				oa.set(i, arr[i]);
+			}
+			return oa;
+		};
+
+	@OpHints(hints = { Simplification.FORBIDDEN })
 	@OpField(names = "focus")
 	public final Function<ObjectArray<Number>, Long[]> longArrFocuser = o -> o
 		.stream().map(l -> l == null ? null : l.longValue()).toArray(Long[]::new);
 
-	@OpHints(hints = {Simplification.FORBIDDEN})
+	@OpHints(hints = { Simplification.FORBIDDEN })
 	@OpField(names = "simplify")
 	public final Function<Float[], ObjectArray<Number>> floatArrSimplifier =
-		f -> new ObjectArray<>(f);
-
-	@OpHints(hints = {Simplification.FORBIDDEN})
+		arr -> {
+			var oa = new ObjectArray<>(Number.class, arr.length);
+			// TODO: Why doesn't System.arraycopy work?
+			for(int i = 0; i < arr.length; i++) {
+				oa.set(i, arr[i]);
+			}
+			return oa;
+		};
+
+	@OpHints(hints = { Simplification.FORBIDDEN })
 	@OpField(names = "focus")
 	public final Function<ObjectArray<Number>, Float[]> floatArrFocuser = o -> o
 		.stream().map(f -> f == null ? null : f.floatValue()).toArray(Float[]::new);
 
-	@OpHints(hints = {Simplification.FORBIDDEN})
+	@OpHints(hints = { Simplification.FORBIDDEN })
 	@OpField(names = "simplify")
 	public final Function<Double[], ObjectArray<Number>> doubleArrSimplifier =
-		d -> new ObjectArray<>(d);
-
-	@OpHints(hints = {Simplification.FORBIDDEN})
+		arr -> {
+			var oa = new ObjectArray<>(Number.class, arr.length);
+			// TODO: Why doesn't System.arraycopy work?
+			for(int i = 0; i < arr.length; i++) {
+				oa.set(i, arr[i]);
+			}
+			return oa;
+		};
+
+	@OpHints(hints = { Simplification.FORBIDDEN })
 	@OpField(names = "focus")
 	public final Function<ObjectArray<Number>, Double[]> doubleArrFocuser = o -> o
```

Comment:
Why `Double[]` and not `double[]`?

Reply:
Uh, no reason, we should have both though, which I have thus added!